examples/apps/aogh/aogh.cpp

```diff
@@ -0,0 +1,201 @@
+#include <FreeRTOS.h>
+#include <task.h>
+#include <string.h>
+
+#include <openthread/ip6.h>
+#include <openthread/joiner.h>
+#include <openthread/openthread-freertos.h>
+#include <openthread/thread.h>
+
+#include <lwip/sockets.h>
+
+#include <nrfx/hal/nrf_gpio.h>
+
+#include "common/code_utils.hpp"
+#include "utils/thin_function.h"
+#include "../../../third_party/openthread/repo/src/core/common/code_utils.hpp"
+
+#define MAC_ADDRESS_SIZE 6
+
+#define CMD_OFF 0x00
+#define CMD_ON 0x01
+#define CMD_BRIGHTNESS 0x02
+#define FIRST_BYTE_OF_MULTICAST 0x48
+
+#define BRIGHTNESS_MIN 0
+#define BRIGHTNESS_MAX 100
+#define TURN_ON 0
+#define TURN_OFF 1
+#define LED_NUM 4
+
+const uint32_t LED_PIN[4] = {13, 14, 15, 16};
+
+static TaskHandle_t sAoghTask;
+
+const char            kTestNetworkName[]  = "Openthread-AoGH";
+const uint16_t        kTestNetworkPanId   = 0xbeef;
+const uint8_t         kTestNetworkChannel = 14;
+const otExtendedPanId kTestNetworkXPanId  = {{0xde, 0xad, 0xff, 0xbe, 0xef, 0xff, 0xca, 0xfe}};
+const otMasterKey     kTestNetworkKey     = {
+    {0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x88, 0x99, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff}};
+
+class AoghTask
+{
+public:
+    AoghTask() {}
+
+    void Initialize(void)
+    {
+        ConnectToThread();
+        InitializeLED();
+    }
+
+    void LaunchServer(void)
+    {
+        AoghServer();
+    }
+
+private:
+    void ConnectToThread(void)
+    {
+        OT_API_CALL(otThreadSetNetworkName(otrGetInstance(), kTestNetworkName));
```

Comment:
use `otrLock();` and `otrUnlock();otrTaskNotifyGive();` for invoking a block of OpenThread APIs?

Reply:
@gjc13 should we put `otrTaskNotifyGive()` inside of `otrUnlock()`?

Reply:
We can 
```
OT_API_CALL(
     otThreadSetNetworkName(...),
     otThreadSetExtendedPanId(...),
.....
);
```